### ISSUE TITLE
Showing Passoword in the place of Email

### ISSUE THREAD

**Issue by VivekS98** [aws-amplify/amplify-ui#5363]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

NextJS

### What browsers are you seeing the problem on?

Firefox

### Which region are you seeing the problem in?

India (Banglore)

### Please describe your bug.

After Signin, a verification step is shown, where instead of email or username it shows password which should never happen. If we are using username for Signup, then it should use another way to verify, but not show the passoword directly.
![Screenshot 2024-07-03 at 13-50-53 RadiXplore v3](https://github.com/aws-amplify/amplify-ui/assets/29965718/0ccc764d-9cc9-4a59-ac29-1eee3df4e263)

### What's the expected behaviour?

If the Auth happens via username, then in the verification step, it must ask for email or phone number for verification. Or there must be a way to disable the verification step.

### Help us reproduce the bug!

Try signing up and signing in through the Amplify UI. After a successful singnin, it shows a verification step where it shows password instead of the email.

### Code Snippet

```typescript
import React from 'react';
import { Amplify } from 'aws-amplify';

import { Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css';

import awsExports from './aws-exports';
Amplify.configure(awsExports);

export default function App() {
  return (
    <Authenticator>
      {({ signOut, user }) => (
        <main>
          <h1>Hello {user.username}</h1>
          <button onClick={signOut}>Sign out</button>
        </main>
      )}
    </Authenticator>
  );
}
```

### Console log output

![Screenshot 2024-07-03 at 13-50-53 RadiXplore v3](https://github.com/aws-amplify/amplify-ui/assets/29965718/d5d84149-927f-4bef-9c36-fa56022b7f6b)

**Comment by esauerbo** [comment]
@VivekS98 I wasn't able to reproduce this with `@aws-amplify/ui-react@6.1.12` (latest). What version are you using? Can you also provide your amplify configuration (minus any sensitive information)?

**Comment by esauerbo** [comment]
@VivekS98 are you still experiencing this issue?

**Comment by VivekS98** [comment]
@esauerbo yes. With the same configuration

**Comment by esauerbo** [comment]
Can you provide the information from my [earlier comment](https://github.com/aws-amplify/amplify-ui/issues/5363#issuecomment-2206375164)? We haven't been able to reproduce this.

**Comment by VivekS98** [comment]
Can you please mention which configuration exactly you need?

**Comment by esauerbo** [comment]
@VivekS98 whatever you're calling Amplify.configure with, which looks like `aws-exports` in your case. And can you tell us what version of `@aws-amplify/ui-react` are you using?

**Comment by thaddmt** [comment]
Is the string that's being returned censored? Like in the screenshot I have posted below? 
<img width="487" alt="Screenshot 2024-07-24 at 10 17 56 AM" src="https://github.com/user-attachments/assets/dff90184-4694-4e7c-92c9-eab7c4357a65">

I tried copying your aws-exports and your sample code and I was not able to reproduce returning a password from the sign up.

I also tried using the `useAuthenticator` hook to print out the `unverifiedUserAttributes` to print out the values, could you try doing that and let us know if password is being printed there? You can try using an app similar to my example posted below.

The following should wrap your application in an AuthenticatorProvider and will use the `useAuthenticator` hook so you can inspect the values as you go through the auth flow

```
function App() {
  const { unverifiedUserAttributes } = useAuthenticator((context) => [context.unverifiedUserAttributes]);
  console.log(JSON.stringify(unverifiedUserAttributes));
  return (
    <Authenticator>
      {({ signOut, user }) => (
        <main>
          <h1>Hello {user.username}</h1>
          <button onClick={signOut}>Sign out</button>
        </main>
      )}
    </Authenticator>
  );
}

export default function ProviderWrappedApp() {
  return (
    <Authenticator.Provider>
      <App />
    </Authenticator.Provider>
  );
}
```

**Comment by reesscot** [comment]
Hi @VivekS98, are you still experiencing this issue?

**Comment by VivekS98** [comment]
Yes

**Comment by VivekS98** [comment]
What happened is this:
I was sent an invite which I used to sign up and create a password. It's that created password which is showing up as email.

**Comment by esauerbo** [comment]
@VivekS98 can you please respond to @thaddmt's comment? https://github.com/aws-amplify/amplify-ui/issues/5363#issuecomment-2248550565 It would be helpful to know whether your password shows up in `unverifiedUserAttributes`.

**Comment by cwomack** [comment]
@VivekS98, wanted to follow up on this issue and see if you're still blocked.  Let us know if you're still in need of assistance on this, and please review the comment above to address the question there! Thanks.

**Comment by VivekS98** [comment]
Hey @cwomack. Thanks for your patience. We had undergone a thorough testing and found out that this was happening since we removed email from mandatry. Maybe this is a configuration issue from our end. I edited the email myself after we created a user management UI in our product. As the co-operation with required people isn't happening (I don't have access to Cognito yet), it's better to close this issue. Thanks.
